lexicon/providers/corenetworks.py

```diff
@@ -0,0 +1,280 @@
+        self._log("Domain %s" % self.domain)
+        self._refresh_auth_file()
+        if 'token' in self.auth_file:
+            self.token = self.auth_file['token']
+            self.expiry = self.auth_file['expiry']
+
+            # Fetch new auth token if expiry is less than 60 seconds away
+            # as it can be we have to wait up to 60 seconds because of
+            # provider rate limiting.
+            if self.expiry-time.time() < 60:
+                self._get_token()
+        else:
+            self._get_token()
+
+        # Store zones for saving one API call
+        self.zones = self._list_zones()
+
+        #Check if requested zone is in zones list
+        zone = next((zone for zone in self.zones if zone["name"] == self.domain), None)
+        if not zone:
+            raise Exception('No domain found like %s.' % self.domain)
+        else:
+            self.domain_id = zone['name']
+        return True
+
+
+    def _list_records(self, rtype=None, name=None, content=None):
+        """List all records. Return an empty list if no records found
+        type, name and content are used to filter records.
+        If possible filter during the query, otherwise filter after response is received.
+        Ref: https://beta.api.core-networks.de/doc/#functon_dnszones_records"""
+        zone = next((zone for zone in self.zones if zone["name"] == self.domain), None)
+        if not zone:
+            raise Exception('Domain not found')
+        query_params = {}
+        if rtype:
+            query_params['type'] = rtype
+        if name:
+            query_params['name'] = self._relative_name(name)
+        if content:
+            query_params['data'] = content
+        payload = self._get("/dnszones/{0}/records/".format(self.domain), query_params)
+        for record in payload:
+            record['content'] = record.pop('data')
+            record['name'] = self._full_name(record['name'])
+            # Core Networks' API does not provide unique IDs for each record
+            # so we generate them ourselves.
+            record['id'] = self._make_identifier( rtype = record['type'], name = record['name'], content = record['content'] )
+        return payload
+
+    def _create_record(self, rtype, name, content):
+        """Creates a record. If record already exists with the same content, do nothing."""
+
+        # Check for existence of record.
+        existing_records = self._list_records(rtype, name, content)
+        new_record_id = self._make_identifier(rtype, self._full_name(name), content)
+        record = next((r for r in existing_records if r["id"] == new_record_id), None)
+        # Nothing to do if true.
+        if record:
+            return True
+
+        data = {
+            'name': self._relative_name(name),
+            'data': content,
+            'type': rtype
+        }
+        if self._get_lexicon_option('ttl'):
+            data['ttl'] = self._get_lexicon_option('ttl')
```

Comment:
I just found out by accident: If the `ttl` is less than 60, this leads to a 

```
(lexicon) bash-3.2$ lexicon corenetworks --auth-username [...] --auth-password [...] create [...] --ttl 59
Traceback (most recent call last):
  File "/Users/ich/.local/share/virtualenvs/lexicon-eYduLJl4/bin/lexicon", line 11, in <module>
    load_entry_point('dns-lexicon', 'console_scripts', 'lexicon')()
  File "/Users/ich/repos/lexicon/lexicon/cli.py", line 117, in main
    results = client.execute()
  File "/Users/ich/repos/lexicon/lexicon/client.py", line 84, in execute
    return self.provider.create_record(record_type, name, content)
  File "/Users/ich/repos/lexicon/lexicon/providers/base.py", line 80, in create_record
    return self._create_record(rtype, name, content)
  File "/Users/ich/repos/lexicon/lexicon/providers/corenetworks.py", line 122, in _create_record
    payload = self._post("/dnszones/{0}/records/".format(self.domain), data)
  File "/Users/ich/repos/lexicon/lexicon/providers/base.py", line 145, in _post
    return self._request('POST', url, data=data, query_params=query_params)
  File "/Users/ich/repos/lexicon/lexicon/providers/corenetworks.py", line 209, in _request
    response.raise_for_status()
  File "/Users/ich/.local/share/virtualenvs/lexicon-eYduLJl4/lib/python2.7/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 415 Client Error: Unsupported Media Type for url: https://beta.api.core-networks.de/dnszones/[...]/records/
```

This limitation doesn't seem [to be documented](https://beta.api.core-networks.de/doc/#functon_dnszones_records_add), but maybe it's still a good idea to intercept the wrong input here.

Reply:
Good catch. I will check for values less than 60 seconds in the next commit.